File: cpp-tiny/lib/Models/Hotspot.cpp


#include "Hotspot.h"

using namespace Tiny;

Hotspot::Hotspot()
{
	delay = float(0);
	name = std::string();
	title = std::string();
	description = std::string();
}

Hotspot::Hotspot(std::string jsonString)
{
	this->fromJson(jsonString);
}

Hotspot::~Hotspot()
{

}

void
Hotspot::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *delayKey = "delay";
    
    if(object.has_key(delayKey)) 
    {
        bourne::json value = object[delayKey];


        
        jsonToValue(&delay, value, "long");


    }

    const char *nameKey = "name";
    
    if(object.has_key(nameKey)) 
    {
        bourne::json value = object[nameKey];


        
        jsonToValue(&name, value, "std::string");


    }

    const char *titleKey = "title";
    
    if(object.has_key(titleKey)) 
    {
        bourne::json value = object[titleKey];


        
        jsonToValue(&title, value, "std::string");


    }

    const char *descriptionKey = "description";
    
    if(object.has_key(descriptionKey)) 
    {
        bourne::json value = object[descriptionKey];


        
        jsonToValue(&description, value, "std::string");


    }


}

bourne::json
Hotspot::toJson()
{
    bourne::json object = bourne::json::object();

    



    object["delay"] = getDelay();


    



    object["name"] = getName();


    



    object["title"] = getTitle();


    



    object["description"] = getDescription();



    return object;

}

long
Hotspot::getDelay()
{
	return delay;
}

void
Hotspot::setDelay(long  delay)
{
	this->delay = delay;
}

std::string
Hotspot::getName()
{
	return name;
}

void
Hotspot::setName(std::string  name)
{
	this->name = name;
}

std::string
Hotspot::getTitle()
{
	return title;
}

void
Hotspot::setTitle(std::string  title)
{
	this->title = title;
}

std::string
Hotspot::getDescription()
{
	return description;
}

void
Hotspot::setDescription(std::string  description)
{
	this->description = description;
}

File: java-vertx/src/main/java/org/openapitools/server/api/model/Hotspot.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.math.BigDecimal;

/**
 * Dictionary containing details of a single hotspot. 
 **/
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Hotspot   {
  
  private BigDecimal delay;
  private String name;
  private String title;
  private String description;

  public Hotspot () {

  }

  public Hotspot (BigDecimal delay, String name, String title, String description) {
    this.delay = delay;
    this.name = name;
    this.title = title;
    this.description = description;
  }

    
  @JsonProperty("delay")
  public BigDecimal getDelay() {
    return delay;
  }
  public void setDelay(BigDecimal delay) {
    this.delay = delay;
  }

    
  @JsonProperty("name")
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

    
  @JsonProperty("title")
  public String getTitle() {
    return title;
  }
  public void setTitle(String title) {
    this.title = title;
  }

    
  @JsonProperty("description")
  public String getDescription() {
    return description;
  }
  public void setDescription(String description) {
    this.description = description;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Hotspot hotspot = (Hotspot) o;
    return Objects.equals(delay, hotspot.delay) &&
        Objects.equals(name, hotspot.name) &&
        Objects.equals(title, hotspot.title) &&
        Objects.equals(description, hotspot.description);
  }

  @Override
  public int hashCode() {
    return Objects.hash(delay, name, title, description);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Hotspot {\n");
    
    sb.append("    delay: ").append(toIndentedString(delay)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    title: ").append(toIndentedString(title)).append("\n");
    sb.append("    description: ").append(toIndentedString(description)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
